tests/load/test_s3.py

```diff
@@ -192,3 +194,25 @@ def test_wait_object_not_exists(path: str, benchmark_time: int) -> None:
         wr.s3.wait_objects_not_exist(file_paths, parallelism=16)
 
     assert timer.elapsed_time < benchmark_time
+
+
+@pytest.mark.parametrize("size", [(5000, 5000), (1, 5000), (5000, 1), (1, 1)])
+def test_wide_df(size, path) -> None:
+    df = pd.DataFrame(np.random.randint(0, 100, size=size))
+    df.columns = df.columns.map(str)
+
+    num_cols = size[0]
+    df["int"] = np.random.choice(["1", "2", None], num_cols)
+    df["decimal"] = np.random.choice(["1.0", "2.0", None], num_cols)
+    df["date"] = np.random.choice(["2020-01-01", "2020-01-02", None], num_cols)
+    df["par0"] = np.random.choice(["A", "B"], num_cols)
+
+    dtype = {
+        "int": "tinyint",
+        "decimal": "double",
+        "date": "date",
+    }
+
+    result = wr.s3.to_csv(df=df, path=path, dataset=True, dtype=dtype, partition_cols=["par0"])
+    partitions_shape = np.array(unwrap_partitions(df)).shape
```

Comment:
Can we add some assertions on the actual numbers of column-axis partitions in the df in this test, please? Or at least assert there's more than one of them

I understand that modin by default will create column partitions if the number of columns is more than 32 but there is no way to tell from the test whether that is actually happening - we might very well work with a single block and now know it